CONTRIBUTING.md

```diff
@@ -0,0 +1,43 @@
+# Issues
+
+Issues have three types: bug, feature request, and support. When reporting a bug, you must include
+the operating system used, any relevant information about the tech stack, and the feature flags
+used. Feature request issues should answer these questions: 
+
+```
+## What would you like implemented? What do you want that Twilight lacks?
+
+## What use case does this request address? This should be more than a single project's use case.
+
+## Are you willing to help towards contributing this feature?
+
+## Is there any other information that we should know?
+```
+
+Before making a support issue or bug report, be sure to consider joining the
+[Discord](https://discord.gg/7jj8n7D) and bringing up your topic in the #support channel.
+
+# Pull Requests
+
+Pull requests must be named with the format `crate: short description of change`, and should use
+lower case letters. If the change spans more than one crate, separate the crate names with a comma
+and a space: `crate1, crate2: short description of change`. Pull requests must be made from a new
+branch. If a PR is made from the `trunk` branch, it will be denied. If adding a feature or
+enhancement, use the term `add` or something sufficiently similar. If fixing a bug, use the term
+`fix`, or something sufficiently similar. Avoid force-pushing to a pull request branch, as this
+erases review comment history.
+
+# Labeling
+
+If you are able, you must label your issues and pull requests appropriately. This includes adding a
+label for each applicable crate, or if the issue/change is project-wide, using `c-all`. `feature`s
+are new additions, and they are distinct from `enhancement`s, which are improvements on existing
+features.  `bugfix`es are self-evident. Any change relating to documentation must use the `docs`
+label. The `discord api` label is used for changes that must be verified against the Discord API for
+correctness.
+
+# Merging
+
+Pull requests require two approvals before merging. They must be merged with the same name as the
+pull request, and as such must follow the naming rules for pull requests. The only possible merge
+option is squash and merge.
```

Comment:
We should add a section on the structure of the repository. ie. the difference between crates in tree and those in a separate repository

```suggestion
option is squash and merge.

# Adding new crates
When a crate is in development or planned to be added, it will not be created in a pull request since that would make it harder to contribute to until it is merged. Instead, new/in development crates will be placed in their own separate repository and will be merged in-tree once the crate is feature complete, tested, and to the standards of the rest of the ecosystem.

# Testing
When changes are added, tests should be added to ensure no regressions will erupt in the new code. It is crucial when implementing a bug-fix that tests are created to ensure that the crate cannot regress and to prevent a new bug from forming from the fix.

# Documentation
When adding new features, or changing an existing user-facing api, documentation should be created or updated respectively and should contain code examples when possible
```

Reply:
I'm not sure about the new crates policy. If I recall correctly, adding new crates has been a case-by-case thing. Lavalink was added with a PR, I'm not sure what happened with the new util crate stuff, and the Oauth crate was a PR but then was moved to a repository due to changes in priorities.

Reply:
I think we should standardize some workflow of adding a crate.

Reply:
Yah, that make sense. I'll resolve this.

Reply:
I have updated the suggestions to remove the info on new crates since it makes sense that there will be standardized flow